projects/aws/image-builder/builder/types.go

```diff
@@ -81,3 +84,17 @@ type NutanixConfig struct {
 	NutanixPassword   string `json:"nutanix_password"`
 	NutanixSubnetName string `json:"nutanix_subnet_name"`
 }
+
+type AMIConfig struct {
+	AMIFilterName       string `json:"ami_filter_name"`
+	AMIRegions          string `json:"ami_regions"`
+	AWSRegion           string `json:"aws_region"`
+	AnsibleExtraVars    string `json:"ansible_extra_vars"`
+	BuilderInstanceType string `json:"builder_instance_type"`
+	CustomRole          string `json:"custom_role"`
+	CustomRoleNames     string `json:"custom_role_names"`
```

Comment:
Maybe we could make this an array? Imagebuilder will take it as space separated collection (I think), but there is not reason we need to expose that to the user, right?
Having an array here will make more clear they can specify more than one custom role

Reply:
Yes you're right, image-builder does take it as a space or comma separated string. The reason I did this is because the user input file is directly used as the packer var-file for image-builder. All we do is unmarshal the input config file, validate the fields and marshal it to the canonical packer var file location  I can see if some transformation in the pipeline is a possible option.

Reply:
To maintain type parity with upstream Imagebuilder, I added another parameter custom_role_names_list which is an array. Users can use both the array field or the space-separated field, but there is a check to ensure they're mutually exclusive. If user provides the list, it is converted to the delimited string format and assigned to the the above field for Marshaling into a format that Packer can use directly with upstream image-builder. I can make it clear in the docs how these fields are supposed to be used.